File: app/src/main/java/com/perc/pavel/sportgeolocationgame/StartGameActivity.java
package com.perc.pavel.sportgeolocationgame;

import android.content.Intent;
import android.graphics.Color;
import android.graphics.Typeface;
import android.location.Location;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.SeekBar;
import android.widget.TextView;
import android.widget.Toast;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class StartGameActivity extends AppCompatActivity {
    
    Profile profile;
    Location location;
    SeekBar sbTeamsNumber;
    TextView tvTeams;
    TextView tvTeamsNumber;
    TextView profileName;
    EditText etRoomId;
    Button btnJoinGame;
    Button btnCreateGame;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_start_game);
        profile = (Profile) getIntent().getSerializableExtra("profile");
        
        sbTeamsNumber = (SeekBar) findViewById(R.id.sbTeamsNumber);
        tvTeams = (TextView) findViewById(R.id.tvTeams);
        tvTeamsNumber = (TextView) findViewById(R.id.tvTeamsNumber);

        profileName = (TextView) findViewById(R.id.profileName);
        profileName.setText(profile.getName());

        etRoomId = (EditText) findViewById(R.id.etRoomId);
        btnJoinGame = (Button) findViewById(R.id.btnJoinGame);
        btnCreateGame = (Button) findViewById(R.id.btnCreateGame);

        Typeface Matiz = Typeface.createFromAsset(getAssets(), "fonts/Matiz.ttf");
        etRoomId.setTypeface(Matiz);
        tvTeams.setTypeface(Matiz);
        tvTeamsNumber.setTypeface(Matiz);
        profileName.setTypeface(Matiz);

        Typeface PhosphateSolid = Typeface.createFromAsset(getAssets(), "fonts/PhosphateSolid.ttf");
        btnJoinGame.setTypeface(PhosphateSolid);
        btnCreateGame.setTypeface(PhosphateSolid);
        
        sbTeamsNumber.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                tvTeamsNumber.setText(String.valueOf(progress + 1));
            }
    
            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
        
            }
    
            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
        
            }
        });
        
    }
    
    public void btnCreateGameClick(View v) {
        List<Integer> teamColorsFull = new ArrayList<>();
    
        teamColorsFull.add(Color.parseColor("#972eff"));// фиолет
        teamColorsFull.add(Color.parseColor("#faed00"));// желтый
        teamColorsFull.add(Color.parseColor("#00a8f3"));// голуб
        teamColorsFull.add(Color.parseColor("#ff7f27"));// оранж
        teamColorsFull.add(Color.parseColor("#0ed145"));// зелен
        teamColorsFull.add(Color.parseColor("#ffaec8"));// розов
        teamColorsFull.add(Color.parseColor("#d81118"));// красн
        teamColorsFull.add(Color.parseColor("#3f48cc"));// синий
        
        Collections.shuffle(teamColorsFull);
        
        final ArrayList<Integer> teamColors = new ArrayList<>(teamColorsFull.subList(0, sbTeamsNumber.getProgress() + 1));
        
        
        
        
        // отправка запроса на создание комнаты
        
        JSONObject json = new JSONObject();
        try {
            json.put("login", profile.getLogin());
            JSONArray ja = new JSONArray();
            
            for (Integer teamColor : teamColors) {
                ja.put(teamColor);
            }
            json.put("colors", ja);
        } catch (JSONException ignored) {}
        
        TcpClient.getInstance().httpPostRequest("create_room", json, new HttpListener() {
            @Override
            public void onResponse(JSONObject message) {
                try {
                    if (message.has("error")) {
                        Toast.makeText(StartGameActivity.this, "Error: " + message.getString("error"), Toast.LENGTH_SHORT).show();
                    }
                    else {
                        Intent intent = new Intent(StartGameActivity.this, GoogleMapsActivity.class);
                        intent.putExtra("profile", profile);
                        intent.putExtra("teamColors", teamColors);
//                        intent.putExtra("myTeamColor", teamColors.get(0));
                        
                        intent.putExtra("roomId", message.getInt("room_id"));
                        intent.putExtra("createGame", true);
                        startActivity(intent);
                    }
                } catch (JSONException e) {
                    Toast.makeText(StartGameActivity.this, "JSONException: " + e, Toast.LENGTH_SHORT).show();
                }
            }
    
            @Override
            public void onFailure(String error) {
                Toast.makeText(StartGameActivity.this, "Failure: " + error, Toast.LENGTH_SHORT).show();
            }
        });
    }
    
    public void btnJoinGameClick(View v) {
        EditText etRoomId = (EditText) findViewById(R.id.etRoomId);
        if (etRoomId.getText().toString().equals("")) {
            Toast.makeText(this, "Номер комнаты не введён.", Toast.LENGTH_SHORT).show();
            return;
        }
        final int roomId = Integer.parseInt(etRoomId.getText().toString());
        
        // отправка запроса на подключение к комнате
        
        JSONObject json = new JSONObject();
        try {
            json.put("login", profile.getLogin());
            json.put("room_id", roomId);
        } catch (JSONException ignored) {}
        
        TcpClient.getInstance().httpPostRequest("connect_to_room", json, new HttpListener() {
            @Override
            public void onResponse(JSONObject message) {
                try {
                    if (message.has("error")) {
                        Toast.makeText(StartGameActivity.this, "Error: " + message.getString("error"), Toast.LENGTH_SHORT).show();
                    }
                    else {
                        ArrayList<Integer> teamColors = new ArrayList<>();
                        JSONArray ja = message.getJSONArray("colors");
                        for (int i = 0; i < ja.length(); i++) {
                            teamColors.add(ja.getInt(i));
                        }
                        
                        Intent intent = new Intent(StartGameActivity.this, GoogleMapsActivity.class);
                        intent.putExtra("profile", profile);
                        intent.putExtra("teamColors", teamColors);
                                                
                        intent.putExtra("roomId", roomId);
                        intent.putExtra("createGame", false);
                        startActivity(intent);
                    }
                } catch (JSONException e) {
                    Toast.makeText(StartGameActivity.this, "JSONException: " + e, Toast.LENGTH_SHORT).show();
                }
            }
        
            @Override
            public void onFailure(String error) {
                Toast.makeText(StartGameActivity.this, "Failure: " + error, Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: app/src/main/java/com/perc/pavel/sportgeolocationgame/activities/StartGameActivity.kt
package com.perc.pavel.sportgeolocationgame.activities

import android.content.Intent
import android.graphics.Color
import android.graphics.Typeface
import android.location.Location
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.perc.pavel.sportgeolocationgame.Profile
import com.perc.pavel.sportgeolocationgame.R
import com.perc.pavel.sportgeolocationgame.serverworking.AbstractHttpClient
import com.perc.pavel.sportgeolocationgame.serverworking.HttpClient
import com.perc.pavel.sportgeolocationgame.serverworking.HttpClientFake

import com.perc.pavel.sportgeolocationgame.serverworking.HttpListener
import kotlinx.android.synthetic.main.activity_start_game.*
import org.jetbrains.anko.alert
import org.jetbrains.anko.sdk25.coroutines.onSeekBarChangeListener
import org.jetbrains.anko.toast

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject
import java.io.Serializable

import java.util.ArrayList


class StartGameActivity : AppCompatActivity() {
    
    internal lateinit var profile: Profile
    internal var location: Location? = null
    
    //    private val httpClient: AbstractHttpClient = HttpClient().inUiThread
    private val httpClient: AbstractHttpClient = HttpClientFake()
    
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_start_game)
        profile = intent.getSerializableExtra("profile") as Profile
        
        profileName.text = profile.name
        
        
        val Matiz = Typeface.createFromAsset(assets, "fonts/Matiz.ttf")
        etRoomId.typeface = Matiz
        tvTeams.typeface = Matiz
        tvTeamsNumber.typeface = Matiz
        profileName.typeface = Matiz
        
        val PhosphateSolid = Typeface.createFromAsset(assets, "fonts/PhosphateSolid.ttf")
        btnJoinGame.typeface = PhosphateSolid
        btnCreateGame.typeface = PhosphateSolid
        
        
        
        
        sbTeamsNumber.onSeekBarChangeListener {
            onProgressChanged { seekBar, progress, fromUser ->
                tvTeamsNumber.text = (progress + 1).toString()
            }
        }

//        sbTeamsNumber.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
//            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
//                tvTeamsNumber.text = (progress + 1).toString()
//            }
//            
//            override fun onStartTrackingTouch(seekBar: SeekBar) {
//                
//            }
//            
//            override fun onStopTrackingTouch(seekBar: SeekBar) {
//                
//            }
//        })
        
    }
    
    fun btnCreateGameClick(v: View) {
        val teamColorsFull = listOf(
                Color.parseColor("#972eff"),// фиолет
                Color.parseColor("#faed00"),// желтый
                Color.parseColor("#00a8f3"),// голуб
                Color.parseColor("#ff7f27"),// оранж
                Color.parseColor("#0ed145"),// зелен
                Color.parseColor("#ffaec8"),// розов
                Color.parseColor("#d81118"),// красн
                Color.parseColor("#3f48cc")// синий
        ).shuffled()
        
        val teamColors = teamColorsFull.take(sbTeamsNumber.progress + 1)
        
        
        pbLoading.visibility = View.VISIBLE
        
        // отправка запроса на создание комнаты
        
        val json = JSONObject()
        
        json.put("login", profile.login)
        val ja = JSONArray()
        
        teamColors.forEach { ja.put(it) }
        
        json.put("colors", ja)
        
        
        httpClient.httpPostRequest("create_room", json, object : HttpListener {
            override fun onResponse(message: JSONObject) {
                pbLoading.visibility = View.GONE
                try {
                    if (message.has("error")) {
                        onFailure(message.getString("error"), "Server returned error")
                        return
                    }
                    
                    val intent = Intent(this@StartGameActivity, GoogleMapsActivity::class.java)
                    intent.putExtra("profile", profile)
                    intent.putExtra("teamColors", teamColors as Serializable)
//                        intent.putExtra("myTeamColor", teamColors.get(0));
                    
                    intent.putExtra("roomId", message.getInt("room_id"))
                    intent.putExtra("createGame", true)
                    startActivity(intent)
                    
                } catch (e: JSONException) {
                    onFailure(e.message!!, "JSONException")
//                    Toast.makeText(this@StartGameActivity, "JSONException: $e", Toast.LENGTH_SHORT).show()
                }
            }
            
            override fun onFailure(error: String, title: String) {
                pbLoading.visibility = View.GONE
                alert(error, title).show()
//                Toast.makeText(this@StartGameActivity, "Failure: $error", Toast.LENGTH_SHORT).show()
            }
        })
    }
    
    fun btnJoinGameClick(v: View) {
        if (etRoomId.text.isEmpty()) {
            toast("Номер комнаты не введён.")
            return
        }
        val roomId = etRoomId.text.toString().toInt()
        
        pbLoading.visibility = View.VISIBLE
        
        // отправка запроса на подключение к комнате
        
        val json = JSONObject()
        json.put("login", profile.login)
        json.put("room_id", roomId)
        
        
        httpClient.httpPostRequest("connect_to_room", json, object : HttpListener {
            override fun onResponse(message: JSONObject) {
                pbLoading.visibility = View.GONE
                
                try {
                    if (message.has("error")) {
                        onFailure(message.getString("error"), "Error from server")
                        return
                    }
                    
                    val teamColors = ArrayList<Int>()
                    val ja = message.getJSONArray("colors")
                    for (i in 0 until ja.length()) {
                        teamColors.add(ja.getInt(i))
                    }
                    
                    val intent = Intent(this@StartGameActivity, GoogleMapsActivity::class.java)
                    intent.putExtra("profile", profile)
                    intent.putExtra("teamColors", teamColors)
                    
                    intent.putExtra("roomId", roomId)
                    intent.putExtra("createGame", false)
                    startActivity(intent)
                    
                } catch (e: JSONException) {
                    onFailure(e.message!!, "JSONException")
//                    Toast.makeText(this@StartGameActivity, "JSONException: $e", Toast.LENGTH_SHORT).show()
                }
            }
            
            override fun onFailure(error: String, title: String) {
                pbLoading.visibility = View.GONE
                alert(error, title).show()
//                Toast.makeText(this@StartGameActivity, "Failure: $error", Toast.LENGTH_SHORT).show()
            }
        })
    }
}
